bootstrap.sh

```diff
@@ -16,20 +16,48 @@
 
 ############################  SETUP PARAMETERS
 app_name='spf13-vim'
-[ -z "$APP_PATH" ] && APP_PATH="$HOME/.spf13-vim-3"
-[ -z "$REPO_URI" ] && REPO_URI='https://github.com/spf13/spf13-vim.git'
-[ -z "$REPO_BRANCH" ] && REPO_BRANCH='3.0'
-debug_mode='0'
-fork_maintainer='0'
-[ -z "$VUNDLE_URI" ] && VUNDLE_URI="https://github.com/gmarik/vundle.git"
 
+if [[ -z "$XDG_CONFIG_HOME" ]]; then
+    [[ -z "$SPF13_VIM" ]] && SPF13_VIM="$HOME/.spf13-vim-3"
+    vim_config_home="$HOME"
+
+    vimrc_before=".vimrc.before"
+    vimrc_before_fork=".vimrc.before.fork"
+    vimrc_bundles=".vimrc.bundles"
+    vimrc_bundles_fork=".vimrc.bundles.fork"
+    vimrc_fork=".vimrc.fork"
+    vimrc_local=".vimrc.local"
+else
+    [[ -z "$SPF13_VIM" ]] && SPF13_VIM="$XDG_CONFIG_HOME/spf13-vim-3"
+    vim_config_home="$XDG_CONFIG_HOME/vim"
+
+    vimrc_before="vimrc.before"
+    vimrc_before_fork="vimrc.before.fork"
+    vimrc_bundles="vimrc.bundles"
+    vimrc_bundles_fork="vimrc.bundles.fork"
+    vimrc_fork="vimrc.fork"
+    vimrc_local="vimrc.local"
+fi
+
+if [[ -z "$XDG_DATA_HOME" ]]; then
+    vim_data_home="$HOME/.vim"
+else
+    # TODO: I can't seem to get vim/vundle etc to play nicely with XDG_DATA_HOME
+    vim_data_home="$HOME/.vim"
+fi
+
+[[ -z "$REPO_URI" ]] && REPO_URI='https://github.com/spf13/spf13-vim.git'
+[[ -z "$REPO_BRANCH" ]] && REPO_BRANCH='3.0'
+debug_mode='1'
+fork_maintainer='0'
+[[ -z "$VUNDLE_URI" ]] && VUNDLE_URI='https://github.com/gmarik/vundle.git'
 ############################  BASIC SETUP TOOLS
 msg() {
     printf '%b\n' "$1" >&2
 }
 
 success() {
-    if [ "$ret" -eq '0' ]; then
+    if [[ "$ret" -eq '0' ]]; then
```

Comment:
neovim is converting to lua, but will still read viml for the foreseeable future.. likely many years.  We're not currently allowed to use lua since not all vims are built with Lua, so that would mean that we'd have to keep two versions around to support both vim and neovim. Sticking with viml is pretty much all we can do if we want to work with regular vim also.